lib/pkcs11h-openssl.c

```diff
@@ -628,6 +636,244 @@ __pkcs11h_openssl_rsa_enc (
 	return rv == CKR_OK ? (int)tlen : -1;
 }
 
+static int
+__pkcs11h_get_pss_params(
+	IN EVP_PKEY_CTX *ctx,
+	IN EVP_PKEY *pkey,
+	IN OUT CK_RSA_PKCS_PSS_PARAMS *params
+)
+{
+	int saltlen;
+	const EVP_MD *md, *mgf1_md;
+
+	_PKCS11H_ASSERT (ctx!=NULL);
+	_PKCS11H_ASSERT (pkey!=NULL);
+	_PKCS11H_ASSERT (params!=NULL);
+
+	if (
+		(EVP_PKEY_CTX_get_signature_md (ctx, &md) <= 0)
+		|| (EVP_PKEY_CTX_get_rsa_mgf1_md (ctx, &mgf1_md) <= 0)
+		|| (EVP_PKEY_CTX_get_rsa_pss_saltlen (ctx, &saltlen) <= 0)
+	) {
+		return -1;
+	}
+
+	if (saltlen == -1) {
+		saltlen = EVP_MD_size (md);
+	}
+	else if (saltlen = -2) {
```

Comment:
From the looks of it, this assignment inside the if-statement was not intended.

Reply:
Thanks for looking at this.
This is the RFC version. This has been split up into two PRs, the first one already merged and second one is PR #34  I haven't closed this as there are some still relevant discussion here.